hudi-client/hudi-client-common/src/main/java/org/apache/hudi/io/HoodieMergeHandle.java

```diff
@@ -370,6 +360,16 @@ public void write(GenericRecord oldRecord) {
     }
   }
 
+  protected void writeToFile(HoodieKey key, GenericRecord avroRecord, boolean shouldPreserveRecordMetadata) throws IOException {
+    if (shouldPreserveRecordMetadata) {
+      // NOTE: `FILENAME_METADATA_FIELD` has to be rewritten to correctly point to the
```

Comment:
Can create handle reuse this method ?

Reply:
Totally. Previously it relied on `preserveMetadata` fields which were not available in `CreateHandle`, so had to bail on moving it to `WriteHandle` class, but then after refactoring forgot to update `CreateHandle`

Reply:
I do want to review all handles more holistically and cleanup quite a bit of duplication and unnecessary complication that we've currently amassed.

Reply:
It is great if you can do that.